R/plot.R

```diff
@@ -780,3 +780,519 @@ make_waterfall_plot <- function(dt_plot,
+#'                                            N_sd_error_bars = 0)
+#'   MSEv_figures$MSEv_bar
+#'   MSEv_figures$MSEv_combination_bar
+#'   MSEv_figures$MSEv_explicand_bar
+#'
+#'   # When there are many combinations or observations, then it can be easier to look at line plots
+#'   MSEv_figures$MSEv_combination_line_point
+#'   MSEv_figures$MSEv_explicand_line_point
+#'
+#'   # We can specify which test observations or combinations to plot
+#'   make_MSEv_eval_crit_plots(explanation_list_named,
+#'                             make_MSEv_comb_and_explicand = TRUE,
+#'                             index_x_explain = c(1, 3:4, 6),
+#'                             N_sd_error_bars = 1)$MSEv_explicand_bar
+#'   make_MSEv_eval_crit_plots(explanation_list_named,
+#'                             make_MSEv_comb_and_explicand = TRUE,
+#'                             id_combination = c(3, 4, 9, 13:15),
+#'                             N_sd_error_bars = 1)$MSEv_combination_bar
+#' }
+#'
+#' @author Lars Henry Berge Olsen
+make_MSEv_eval_crit_plots <- function(explanation_list,
+                                      index_x_explain = NULL,
+                                      id_combination = NULL,
+                                      geom_col_width = 0.9,
+                                      N_sd_error_bars = 1,
+                                      make_MSEv_comb_and_explicand = FALSE,
+                                      ...) {
+
+  # Setup and checks ----------------------------------------------------------------------------
+  if (!requireNamespace("ggplot2", quietly = TRUE)) {
+    stop("ggplot2 is not installed. Please run install.packages('ggplot2')")
+  }
+
+  # Ensure that even a single explanation object is in a list
+  if ("shapr" %in% class(explanation_list)) explanation_list <- list(explanation_list)
+
+  # Name the elements in the explanation_list if no names have been provided
+  if (is.null(names(explanation_list))) explanation_list = MSEv_name_explanation_list(explanation_list)
+
+  # Check that the explanation objects explain the same observations
+  MSEv_check_explanation_list(explanation_list)
+
+  # Get the number of observations and combinations
+  n_explain <- explanation_list[[1]]$internal$parameters$n_explain
+  n_combinations <- explanation_list[[1]]$internal$parameters$n_combinations
+
+  # Create data.tables of the MSEv values
+  MSEv_dt_list = MSEv_extract_MSEv_values(explanation_list = explanation_list,
+                                          index_x_explain = index_x_explain,
+                                          id_combination = id_combination)
+  MSEv_dt = MSEv_dt_list$MSEv
+  MSEv_explicand_dt = MSEv_dt_list$MSEv_explicand
+  MSEv_combination_dt = MSEv_dt_list$MSEv_combination
+
+  # Warnings related to the error bars
+  if (!is.null(N_sd_error_bars) && N_sd_error_bars > 0) {
+    if (n_explain < 30) {
+      message(sprintf("The error bars might be wide as the number of observations to explain is only %d.\n",
```

Comment:
not sure I follow. So what you are plotting is sd/sqrt(n)? clarify in documentation

Reply:
I have added clarifying text in the documentation of the MSEv plot function.

Reply:
I have also added some sentences to the vignette